controllers/imagerepository_controller.go

```diff
@@ -230,6 +234,139 @@ func (r *ImageRepositoryReconciler) Reconcile(ctx context.Context, req ctrl.Requ
 	return ctrl.Result{}, nil
 }
 
+func (r *ImageRepositoryReconciler) NotificationExistsInQuayByUUID(UUID string, imageRepository *imagerepositoryv1alpha1.ImageRepository) (quay.Notification, error) {
+	notification := quay.Notification{}
+	allNotifications, err := r.QuayClient.GetNotifications(r.QuayOrganization, imageRepository.Spec.Image.Name)
+	if err != nil {
+		return notification, err
+	}
+	for _, quayNotification := range allNotifications {
+		if quayNotification.UUID == UUID {
+			notification = quayNotification
+			break
+		}
+	}
+
+	return notification, nil
+}
+
+func (r *ImageRepositoryReconciler) NotificationExistsInQuayByTitle(Title string, imageRepository *imagerepositoryv1alpha1.ImageRepository) (quay.Notification, error) {
+	notification := quay.Notification{}
+	allNotifications, err := r.QuayClient.GetNotifications(r.QuayOrganization, imageRepository.Spec.Image.Name)
+	if err != nil {
+		return notification, err
+	}
+	for _, quayNotification := range allNotifications {
+		if quayNotification.Title == Title {
+			notification = quayNotification
+			break
+		}
+	}
+
+	return notification, nil
+}
+
+func (r *ImageRepositoryReconciler) HandleNotifications(ctx context.Context, imageRepository *imagerepositoryv1alpha1.ImageRepository) error {
+	log := ctrllog.FromContext(ctx).WithName("HandleNotifications")
+
+	if imageRepository.Status.Notifications == nil {
+		// No status notifications to check
+		return nil
+	}
+	log.Info("Handling notifications")
+	for index := 0; index < len(imageRepository.Status.Notifications); index++ {
+		statusNotification := &imageRepository.Status.Notifications[index] // This way we can update the UUID if notification gets updated
+		existsInSpec := false
+		for _, notification := range imageRepository.Spec.Notifications {
+			if notification.Title == statusNotification.Title {
+				existsInSpec = true
+				quayNotification, err := r.NotificationExistsInQuayByUUID(statusNotification.UUID, imageRepository)
+				if err != nil {
+					return err
+				}
+				if quayNotification.UUID != "" && (quayNotification.Title != notification.Title || quayNotification.Event != string(notification.Event) || quayNotification.Method != string(notification.Method) || quayNotification.Config.Url != notification.Config.Url) {
+					// Updated item in Spec.Notifications: update notification in Quay
+					log.Info("Updating notification in Quay", "Title", notification.Title, "Event", notification.Event, "Method", notification.Method)
+					updatedNotification, err := r.QuayClient.UpdateNotification(
```

Comment:
This is really similar to what's done in `AddNotification`

Reply:
Note that here we are **Updating** the notification, which is a different operation to **Add** the notification, although the calls could look similar